### PR TITLE
db/context-menu-button

### PR DESCRIPTION
**Pull request by karolkolodziej**
Fixed context menu button click.

### CODE REVIEW

File: ts/Dashboards/EditMode/EditContextMenu.ts [hunk 1]
```diff
@@ -47,21 +47,7 @@ class EditContextMenu extends Menu {
...
-            },
-            langKey: 'editMode',
-            events: {
-                click: function (this: MenuItem): void {
-                    (this.menu as EditContextMenu).editMode.onEditModeToggle();
-                }
-            }
-        }
-    });
+    public static items = merge(Menu.items, {});
```
Comment (sebastianbochan) [inline ts/Dashboards/EditMode/EditContextMenu.ts:50]: Do we need merge here?

File: ts/Dashboards/EditMode/EditRenderer.ts [hunk 2]
```diff
@@ -486,6 +482,14 @@ function renderText(
             {},
             parentElement
         );
+
+        if (options.click) {
+            textElem.addEventListener('click', (): void => {
+                if (options.click instanceof Function) {
+                    options.click();
```
Comment (sebastianbochan) [inline ts/Dashboards/EditMode/EditRenderer.ts:489]: When the toggle is created, the click event is also added. So, the click event will be triggered twice on the toggle button.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

**Review by bre1470 — COMMENTED** [review]
If the button is supposed to work out of the box, then it fails for me in Safari on the `dashboards/demos/layout` demo.

File: ts/Dashboards/EditMode/EditContextMenu.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/Dashboards/EditMode/EditContextMenu.ts:50]: `merge` create a copy, IMO it's safer to work on a fresh copy than accidentally modify an existing object.

File: ts/Dashboards/EditMode/EditRenderer.ts (hunk 2 above)
Comment (karolkolodziej) [reply ts/Dashboards/EditMode/EditRenderer.ts:489]: Tested this and everything works the same as in `dashboards/main`. 
When I tried to simplify it the toggle in the sidebar fail. I guess it needs to stay as it is.

File: ts/Dashboards/EditMode/EditRenderer.ts [hunk 3]
```diff
@@ -486,6 +482,14 @@ function renderText(
...
+                }
+            });
```
Comment (pawellysy) [inline ts/Dashboards/EditMode/EditRenderer.ts:491]:
```suggestion
        if (options.click && (options.click instanceof Function)) {
            textElem.addEventListener('click', options.click);
```

**Review by pawellysy — CHANGES_REQUESTED** [review]
Minor polishing

**Pull request closed without merging**